src/setup.ts

```diff
@@ -51,6 +53,64 @@ const labelNodes = () => {
 
 const setup = () => window.onload = () => {
     labelNodes();
+    let comments: Promise<NTRComment[]> = getComments();
+    let lastSelected: HTMLElement | undefined = undefined;
+    let lastSelectedBackground: string = "";
+
+    comments.then((com: NTRComment[]) => {
+        com.forEach(element => {
+            let elem: HTMLElement | null = document.getElementById(element.nodeID);
+            let i: number = 0;
+
+            // doesn't work very well for displaying more than one comment and i don't really know why
+            if (elem) {
+                elem.innerHTML = elem.innerHTML.substring(0, element.startOffset) +
+                    "<span id=comment_text_" + i + " onMouseEnter=\"document.getElementById(&quot;comment_" + i +
+                    "&quot;).style.display = &quot;block&quot;\" onMouseLeave=\"document.getElementById(&quot;comment_" + i +
+                    "&quot;).style.display = &quot;none&quot;\">" +
+                    elem.innerHTML.substring(element.startOffset, element.endOffset) +
+                    "</span><div id=comment_" + i + " style=\"background-color: none; display: none; text-align: center;\">" +
+                    element.text + "</div>" +
+                    elem.innerHTML.substring(element.endOffset);
+
+                    let child: HTMLElement | null = document.getElementById("comment_text_" + i);
+
+                    if (child) {
+                        child.style.backgroundColor = "yellow";
+                    }
+
+                    i++;
+            }
+        })
+    })
+
+    document.addEventListener("selectionchange", () => {
+        // This is all a bunch of temporary code showing what can be done with selections and ranges.
+        // Because of this, the style is pretty awful. We should discuss exactly what we want to do with 
+        // selections and what format we want the data in before we proceed too much further.
+        const selection: Selection | null = document.getSelection();
+        if (selection == null || selection.rangeCount === 0) {
+            if (lastSelected) {
+                lastSelected.style.backgroundColor = lastSelectedBackground;
+            }
+            return;
+        }
+        const range: Range = selection.getRangeAt(0);
+        /*
+         Do things with selection here
+         https://developer.mozilla.org/en-US/docs/Web/API/Selection
+         https://developer.mozilla.org/en-US/docs/Web/API/Range
+         */
+        const node: Node | null = range.startContainer.parentNode;
+        if (node) {
+            if (lastSelected) {
+                lastSelected.style.backgroundColor = lastSelectedBackground;
+            }
+            lastSelected = (node as HTMLElement);
+            lastSelectedBackground = (node as HTMLElement).style.backgroundColor;
+            (node as HTMLElement).style.backgroundColor = "red";
+        }
+    });
```

Comment:
This was temporary example code removed in last merge to main. Please make sure it stays deleted.